packages/graphql/lib/src/utilities/helpers.dart

```diff
@@ -36,14 +36,14 @@ Map<String, dynamic> _recursivelyAddAll(
   Map<String, dynamic> target,
   Map<String, dynamic> source,
 ) {
-  target = unwrapMap(target);
+  target = Map.from(unwrapMap(target));
```

Comment:
good catch - I though prepending an empty literal in `deeplyMergeLeft` would do this, but it doesn't because of the deep traversal.